src/SharpLearning.Backend.Cntk.Test/CntkBackendTest.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SharpLearning.Backend.Cntk.Test
+{
+    [TestClass]
+    public class CntkBackendTest
+    {
+        [TestMethod]
+        public void CntkBackendTest_CtorAndDispose()
+        {
+            // Doesn't really do anything here
+            using (var b = new CntkBackend(DeviceType.Automatic))
+            {
+            }
+        }
+
+        [TestMethod]
+        public void CntkBackendTest_CreateGraph()
+        {
+            var deviceType = DeviceType.Automatic;
+            using (var b = new CntkBackend(deviceType))
+            using (var g = b.CreateGraph())
+            {
+                Assert.AreEqual(g.DefaultDeviceType, deviceType);
+            }
+        }
+
+        [TestMethod]
+        public void CntkBackendTest_CreateGraph_Placeholder()
+        {
+            var deviceType = DeviceType.Automatic;
+            using (var b = new CntkBackend(deviceType))
+            using (var g = b.CreateGraph())
+            {
+                g.Placeholder(DataType.Single, new int[] { 1, 2, 3 }, "p");
```

Comment:
This fails with:
```
Message: Test method SharpLearning.Backend.Cntk.Test.CntkBackendTest.CntkBackendTest_CreateGraph_Placeholder threw exception: 
System.TypeInitializationException: The type initializer for 'CNTK.CNTKLibPINVOKE' threw an exception. ---> System.TypeInitializationException: The type initializer for 'SWIGExceptionHelper' threw an exception. ---> System.DllNotFoundException: Unable to load DLL 'Cntk.Core.CSBinding-2.3.1.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
```
@mdabros have you seen this before? The nuget package has been added to the project...

Reply:
The dll can be found at:
```
SharpLearning\packages\CNTK.GPU.2.3.1\support\x64\Release
```
but it is not copied to output.

Reply:
The CNTK package definition is really weird... seems to be ill defined.

Reply:
No, I have not experienced that error before. But I do know that there are some issues with the current CNTK package: https://github.com/Microsoft/CNTK/issues/2527 and https://github.com/Microsoft/CNTK/issues/2807

However, I am not sure if that relates to the error above.

Reply:
As discussed, I tried changing the `SharpLearning.Backend.Cntk.Test` test project to .net framework to see if this was causing the issue. Sadly, this still result in the same error.

Reply:
It seems that the only way to get this running is to add a x64 solution platform, and run the test using this specific platform. Using anycpu together with x64 test settings is not enough. 

I guess this ties back to the packaging you mentioned earlier. This might also be another indication that the backend projects should have their own solution.

Reply:
@mdabros please do not add x64 to solution platform, that is not necessary.

Reply:
@mdabros no doubt this is because the Cntk `targets` file does some logic on the Platform attribute of the project. So basically, you just have to change the Cntk.Test project from AnyCPU to x64, you can try and make a new branch from `backends` and create a PR back to it again. The devil lies in trying not to mess all project files up, which can quickly become a mess ;)

Reply:
You don't need to change the Solution platform in order to do this. We might still consider whether we should in fact have just a x64 solution format since a lot is tied into this. But hopefully Cntk will become more cross-plat over time. And TF is.

Reply:
@nietras Good warning, I will change only the Cntk.Test project to x64. On a side note, it seems we still have to change the Cntk.Test project format to .net framework. If not, it is not possible to use CNTK directly from the project. I think this related to the two issues I linked to above.